shared/Microsoft.Extensions.Buffers.Sources/MemoryPoolBlock.cs

```diff
@@ -12,12 +12,12 @@ namespace System.Buffers
     /// Block tracking object used by the byte buffer memory pool. A slab is a large allocation which is divided into smaller blocks. The
     /// individual blocks are then treated as independent array segments.
     /// </summary>
-    internal class MemoryPoolBlock : OwnedMemory<byte>
+    internal class MemoryPoolBlock : MemoryManager<byte>
     {
         private readonly int _offset;
         private readonly int _length;
-        private int _referenceCount;
-        private bool _disposed;
+
+        private int _pinCount;
```

Comment:
Remove, the only thing we want is the disposal check.

Reply:
Why not have this if it's freeish?

Reply:
It's unless for us, we don't care about counting pins and unpins. It's code I don't want to worry about.

Reply:
It can help test our consumer implementations .

Reply:
I don't understand why you want to keep it, can we just remove it. Like kill it, the thing we want to keep is the disposal check, not this crap check.